src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -156,6 +180,13 @@ class ReferralService(
     throw AccessError(user, "unsupported user type", listOf("logins from ${user.authSource} are not supported"))
   }
 
+  private fun getSentReferralSummaryForServiceProviderUser(user: AuthUser, sentReferralFilterSpecification: Specification<ReferralSummary>, page: Pageable?): Iterable<ReferralSummary> {
+    // todo: query for referrals where the service provider has been granted nominated access only
+    val filteredSpec = referralAccessFilter.serviceProviderReferrals(sentReferralFilterSpecification, user)
+    return if (page == null) referralSummariesRepository.findAll(filteredSpec).sortedBy { it.sentAt } else referralSummariesRepository.findAll(filteredSpec, page)
```

Comment:
in this method i don't think page should be nullable? this would simplify the code on line 186.

Reply:
Sure, got your point